core/src/main/codegen/templates/Parser.jj

```diff
@@ -4528,17 +4528,50 @@ SqlLiteral DateTimeLiteral() :
         return SqlParserUtil.parseTimestampLiteral(p, s.end(this));
     }
 |
-    <DATE> { s = span(); } <QUOTED_STRING> {
-        return SqlParserUtil.parseDateLiteral(token.image, s.end(this));
-    }
+    <DATE> { s = span(); } ((<QUOTED_STRING> | <BIG_QUERY_QUOTED_STRING>)
+      {
+          p = SqlParserUtil.stripQuotes(token.image, "'", "'", "\\'",
+              Casing.UNCHANGED);
+          return SqlParserUtil.parseDateLiteralWithoutQuote(p, s.end(this));
+      }
+     | <BIG_QUERY_DOUBLE_QUOTED_STRING>
```

Comment:
Why do we need this BIG_QUERY_DOUBLE_QUOTED_STRING case?

Reply:
Oh I see. You opened a '(' several lines back and didn't indent. Confusing.

I suggest you add a rule (function) that matches <QUOTED_STRING>, <BIG_QUERY_QUOTED_STRING> and <BIG_QUERY_DOUBLE_QUOTED_STRING> and use it in all of these literals.

Reply:
Will do

Reply:
@julianhyde Updated based on your feedback. Let me know whether it's what you expected.